ckanext/switzerland/logic.py

```diff
@@ -469,14 +469,40 @@ def ogdch_user_create(context, data_dict):
     """overwrites the core user creation to send an email
     to new users"""
     user = core_user_create(context, data_dict)
-    tk.get_action('ogdch_add_users_to_groups')(context, {'user_id': user['id']})  # noqa
-    send_email_on_registration = config.get('ckanext.switzerland.send_email_on_user_registration', True)  # noqa
-    if send_email_on_registration and user.get('email'):
-        try:
-            send_registration_email(user)
-            h.flash_success("An email has been send to the user {} at {}."  # noqa
+    tk.get_action('ogdch_add_users_to_groups')(
+        context, {'user_id': user['id']}
+    )
+    send_email_on_registration = tk.asbool(config.get(
+        'ckanext.switzerland.send_email_on_user_registration', True
+    ))
+
+    if not(send_email_on_registration and user.get('email')):
+        return user
+
+    success = False
+    exception = ''
+    try:
+        send_registration_email(user)
+        success = True
+    except Exception as e:
+        exception = e
+
+    try:
+        if success:
```

Comment:
Wouldn't it be better to specify directly success to be equal True?

Reply:
I'm not sure what you mean. `success` is set to `True`, on line 486, after registration email has been successfully sent.